### PR TITLE
Use LayoutService#getAll/ActiveLayouts

### PR DESCRIPTION
**Pull request by msladek**
https://synjira.atlassian.net/browse/CELDEV-1074

### CODE REVIEW

File: src/main/webapp/templates/celTemplates/PageLayouts.vm [hunk 1]
```diff
@@ -1,6 +1,5 @@
 #macro(showListing)
-  #set($isLayoutEditorAvailable = $celementsweb.layoutEditorAvailable())
-  #if(!$isLayoutEditorAvailable)
+  #if(!$services.layout.layoutEditorAvailable())
     <p>The celements LayoutEditor is missing. Please get a current version from
       <a target="_blank" href="http://maven.celements.ch/releases/celLayoutEditor/?C=M;O=D">http://maven.celements.ch/releases/celLayoutEditor/</a>
       and install it with the <a href="$xwiki.getURL('XWiki.Import','import','xpage=overlay&conf=XarImport')">Import Manager</a>.
```
Comment (fpichler) [inline src/main/webapp/templates/celTemplates/PageLayouts.vm]: can you please replace deprecated usage of xwiki.getURL with services.url.getURL ?

File: src/main/webapp/templates/celTemplates/PageLayouts.vm [hunk 2]
```diff
@@ -9,38 +8,48 @@
     <p>click Pretty Name to edit</p>
     <p>Import additional Layouts with the <a href="$xwiki.getURL('XWiki.Import','import','xpage=overlay&conf=XarImport')">Import Manager</a>.
```
Comment (fpichler) [inline src/main/webapp/templates/celTemplates/PageLayouts.vm]: can you please replace deprecated usage of xwiki.getURL with services.url.getURL ?

File: src/main/webapp/templates/celTemplates/PageLayouts.vm [hunk 3]
```diff
@@ -9,38 +8,48 @@
...
         <tr>
-          #set($pretty_name = $layoutsMap.get($layoutKey))
-          #set($plApi = $celementsweb.getPageLayoutApiForName($layoutKey))
+          <!-- layout: $layout -->
           <td class="column${velocityCount}">
-            #if($isLayoutEditorAvailable)
-              #set($editLink = $xwiki.getURL("$!{layoutKey}.WebHome", 'view'))
-              <a href="$editLink">$!pretty_name</a>
+            #if($services.layout.layoutEditorAvailable())
+              <a href="$!services.url.getExternalURL($layout.spaceRef)">
```
Comment (fpichler) [inline src/main/webapp/templates/celTemplates/PageLayouts.vm]: why getExternalURL? getURL should be sufficient.

File: src/main/webapp/templates/celTemplates/PageLayouts.vm [hunk 4]
```diff
@@ -9,38 +8,48 @@
...
-                <img src="$celementsweb.getSkinFile(':icons/silk/cross.gif', 'file')" style="border: 0px;" alt="delete"/></a>
-              </td>
+          <td>$!layout.isActive()</td>
+          #if($isAdmin)
+            #set($layoutFN = $services.reference.serialize($layout.spaceRef))
+            <td><a href="?xpage=overlay&conf=PageLayoutsListing&action=export&layout=$!layoutFN">export</a></td>
+            #if($layoutWiki == $context.database)
+            <td>
+              <a href="?xpage=overlay&conf=PageLayoutsListing&action=delete&layout=$!layoutFN">
+              <img src="$celementsweb.getSkinFile(':icons/silk/cross.gif', 'file')" style="border: 0px;" alt="delete"/></a>
```
Comment (fpichler) [inline src/main/webapp/templates/celTemplates/PageLayouts.vm]:
```suggestion
              <img src="$services.celementsweb.getSkinFile(':icons/silk/cross.gif', 'file')" style="border: 0px;" alt="delete"/></a>
```

File: src/main/webapp/templates/celTemplates/attachmentsinline.vm [hunk 5]
```diff
@@ -63,22 +63,25 @@ document.observe("load", function() { new XWiki.viewers.Attachments(); });
...
+    <div><label class="hidden" for="xwikiuploadfile">$msg.get('core.viewers.attachments.upload.file')</label><input id="xwikiuploadfile" type="file" name="filepath" value="" size="40" class="uploadFileInput"/></div>
+    <div>
+    <span class="buttonwrapper"><input type="submit" value="$msg.get('core.viewers.attachments.upload.submit')" class="button"/></span>
+    <span class="buttonwrapper"><input type="reset" value="$msg.get('core.viewers.attachments.upload.cancel')" class="button"/></span>
+    </div>
+    </fieldset>
+  </div>
+</form>
+#if($doc.isNew())
+  $doc.save() ## avoid uploading stuff on new docs since UploadAction sucks balls
```
Comment (fpichler) [inline src/main/webapp/templates/celTemplates/attachmentsinline.vm]: this leads to a drive by creation of empty documents. What issues do you have with UploadAction?

**Review by fpichler — CHANGES_REQUESTED** [review]

File: src/main/webapp/templates/celTemplates/PageLayouts.vm (hunk 3 above)
Comment (msladek) [reply src/main/webapp/templates/celTemplates/PageLayouts.vm]: the space may now be a central which is a different domain.

File: src/main/webapp/templates/celTemplates/attachmentsinline.vm (hunk 5 above)
Comment (msladek) [reply src/main/webapp/templates/celTemplates/attachmentsinline.vm]: The `UploadAction` appearently wasn't written with the intention to correctly create docs if they don't exist.
I've already fixed many docs without default lang only being accessed via Upload-Overlay.
Thus this makes sure the doc exists beforehand. Rarely will the upload page be opened on new docs and be cancelled. These rare cases of empty docs is preferrable to many cases of docs without sane defaults.

Edit: Created issue `CELDEV-1077` for adressing the UploadAction and added TODO comment for future removal

File: src/main/webapp/templates/celTemplates/PageLayouts.vm (hunk 3 above)
Comment (fpichler) [reply src/main/webapp/templates/celTemplates/PageLayouts.vm]: 🆗

File: src/main/webapp/templates/celTemplates/PageLayouts.vm (hunk 4 above)
Comment (fpichler) [reply src/main/webapp/templates/celTemplates/PageLayouts.vm]: please add commit or comment

**Review by fpichler — CHANGES_REQUESTED** [review]

**Review by fpichler — APPROVED** [review]

**Pull request merged**